var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -39,6 +39,16 @@ class M4(Package):
 
     depends_on('libsigsegv', when='+sigsegv')
 
+    def _make_executable(self, name):
+        return Executable(join_path(self.prefix.bin, name))
+
+    def setup_dependent_package(self, module, dependent_spec):
+        # m4 is very likely to be a build dependency, so we add the
```

Comment:
For what is worth I added the `setup_dependent_environment` in `autoconf` to do exactly what @adamjstewart suggests. The use case were packages that came with `configure.ac` and `Makefile.am`, but without a properly generetaed `configure` script.